cli/azd/pkg/containerregistry/remote_build.go

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package containerregistry
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerregistry/armcontainerregistry"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blockblob"
+	"github.com/azure/azure-dev/cli/azd/internal/tracing"
+	"github.com/azure/azure-dev/cli/azd/internal/tracing/events"
+	"github.com/azure/azure-dev/cli/azd/pkg/account"
+	"github.com/azure/azure-dev/cli/azd/pkg/httputil"
+)
+
+// RemoteBuildManager provides functionality to interact with the Azure Container Registry Remote Build feature.
+type RemoteBuildManager struct {
+	credentialProvider account.SubscriptionCredentialProvider
+	httpClient         httputil.HttpClient
+	armClientOptions   *arm.ClientOptions
+}
+
+// UploadBuildSource uploads the build source to the specified registry. The returned SourceUploadDefinition may be used
+// when scheduling a build to reference the uploaded source.
+func (r *RemoteBuildManager) UploadBuildSource(
+	ctx context.Context,
+	subscriptionID string,
+	resourceGroupName string,
+	registryName string,
+	buildSourcePath string,
+) (armcontainerregistry.SourceUploadDefinition, error) {
+	ctx, span := tracing.Start(ctx, events.RemoteBuildUploadEvent)
```

Comment:
Just curious about the motivation behind tracking the upload as well here.

Reply:
Didn't have a great rationale here, and I suspect I will just remove this stuff based on our internal discussion and move to just incrementing a count per remote build.  Back when I was on the "span plan" for this stuff it felt like "the time spent uploading the build context" was going to be something that was generally useful (just because it's sort of a large portion of the execution time that we don't really control) and carving it out would be useful, especially if you were to look at an end to end trace of a single command (this is sort of just my "span all the things" from a distributed tracing point of view).  But I'm not wed to keeping this if we don't think it provides helpful data.

Reply:
FWIW - I ended up deleting this (and the span stuff in the RunDockerBuildRequestWithLogs method) in favor of just doing a call in RunDockerBuildRequestWithLogs to `tracing.IncrementUsageAttribute(fields.RemoteBuildCount.Int(1))` which seemed to match other uses of `IncrementUsageAttribute` across the codebase.